File: hospital-ai/backend/app/schemas/risk.py
"""Pydantic schemas for Intake Stage 5 — Patient Risk Profiling."""

from datetime import datetime
from typing import Any, Dict, List, Optional
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.registration import ProcessingJobOut


class RiskStartRequest(BaseModel):
    job_id: UUID


class CategoryRiskOut(BaseModel):
    name: str
    level: str
    score: float
    reason: str
    evidence: List[str] = Field(default_factory=list)
    confidence: float = 0.7
    factors: List[str] = Field(default_factory=list)


class RiskAlertOut(BaseModel):
    severity: str
    message: str
    category: Optional[str] = None


class RiskProfileOut(BaseModel):
    id: UUID
    processing_job_id: UUID
    patient_id: UUID
    overall_level: str
    overall_score: float
    overall_confidence: Optional[float] = None
    categories_json: List[Any] = Field(default_factory=list)
    top_risk_factors_json: List[Any] = Field(default_factory=list)
    important_findings_json: List[Any] = Field(default_factory=list)
    alerts_json: List[Any] = Field(default_factory=list)
    distribution_json: Dict[str, Any] = Field(default_factory=dict)
    timeline_json: List[Any] = Field(default_factory=list)
    disclaimer: Optional[str] = None
    processing_time_ms: Optional[int] = None
    status: str = "Completed"
    error_message: Optional[str] = None
    created_at: datetime


class RiskStartResponse(BaseModel):
    job_id: UUID
    patient_id: UUID
    status: str
    current_stage: str
    next_stage: str = "Patient Knowledge Graph"
    overall_level: str
    overall_score: float
    overall_confidence: float
    processing_time_ms: int
    categories: List[CategoryRiskOut] = Field(default_factory=list)
    top_risk_factors: List[str] = Field(default_factory=list)
    important_findings: List[str] = Field(default_factory=list)
    alerts: List[RiskAlertOut] = Field(default_factory=list)
    distribution: Dict[str, int] = Field(default_factory=dict)
    timeline: List[Dict[str, Any]] = Field(default_factory=list)
    disclaimer: str
    warnings: List[str] = Field(default_factory=list)
    risk_profile: RiskProfileOut
    processing_job: ProcessingJobOut
    patient_context_version: int = 1


class RiskStatusResponse(BaseModel):
    job_id: UUID
    status: str
    current_stage: str
    next_stage: str
    progress_pct: int
    risk_completed: bool = False
    overall_level: Optional[str] = None
    overall_score: Optional[float] = None
    error_message: Optional[str] = None
    processing_job: ProcessingJobOut

File: hospital-ai/frontend/src/types/risk.ts
/** Intake Stage 5 — Patient Risk Profiling types. */

export type RiskLevel = 'Very Low' | 'Low' | 'Moderate' | 'High' | 'Critical';

export interface CategoryRisk {
  name: string;
  level: RiskLevel | string;
  score: number;
  reason: string;
  evidence: string[];
  confidence: number;
  factors: string[];
}

export interface RiskAlert {
  severity: 'info' | 'warning' | 'critical' | string;
  message: string;
  category?: string | null;
}

export interface RiskProfile {
  id: string;
  processing_job_id: string;
  patient_id: string;
  overall_level: string;
  overall_score: number;
  overall_confidence?: number | null;
  categories_json: CategoryRisk[];
  top_risk_factors_json: string[];
  important_findings_json: string[];
  alerts_json: RiskAlert[];
  distribution_json: Record<string, number>;
  timeline_json: Array<Record<string, unknown>>;
  disclaimer?: string | null;
  processing_time_ms?: number | null;
  status: string;
  error_message?: string | null;
  created_at: string;
}

export interface RiskStartResult {
  job_id: string;
  patient_id: string;
  status: string;
  current_stage: string;
  next_stage: string;
  overall_level: string;
  overall_score: number;
  overall_confidence: number;
  processing_time_ms: number;
  categories: CategoryRisk[];
  top_risk_factors: string[];
  important_findings: string[];
  alerts: RiskAlert[];
  distribution: Record<string, number>;
  timeline: Array<Record<string, unknown>>;
  disclaimer: string;
  warnings: string[];
  risk_profile: RiskProfile;
  processing_job: { id: string; current_stage: string; status: string };
  patient_context_version: number;
}

export interface RiskStatus {
  job_id: string;
  status: string;
  current_stage: string;
  next_stage: string;
  progress_pct: number;
  risk_completed: boolean;
  overall_level?: string | null;
  overall_score?: number | null;
  error_message?: string | null;
}

export const RISK_LEVEL_TONE: Record<
  string,
  'green' | 'blue' | 'amber' | 'red' | 'gray'
> = {
  'Very Low': 'green',
  Low: 'green',
  Moderate: 'amber',
  High: 'red',
  Critical: 'red',
};
